packages/api/db/migration/20240115182150_create_animal_breed_tables.js

```diff
@@ -0,0 +1,109 @@
+/*
+ *  Copyright (c) 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+export const up = async function (knex) {
+  await knex.schema.createTable('default_animal_breed', (table) => {
+    table.increments('id').primary();
+    table.integer('default_type_id').notNullable().references('id').inTable('default_animal_type');
+    table.string('key').notNullable();
+  });
+
+  await knex.schema.createTable('custom_animal_breed', (table) => {
+    table.increments('id').primary();
+    table.uuid('farm_id').notNullable().references('farm_id').inTable('farm');
+    table.integer('default_type_id').references('id').inTable('default_animal_type');
+    table.integer('custom_type_id').references('id').inTable('custom_animal_type');
+    table.string('breed').notNullable();
+    table.boolean('deleted').notNullable().defaultTo(false);
+    table.string('created_by_user_id').references('user_id').inTable('users');
+    table.string('updated_by_user_id').references('user_id').inTable('users');
+    table.dateTime('created_at').notNullable().defaultTo(new Date('2000/1/1').toISOString());
+    table.dateTime('updated_at').notNullable().defaultTo(new Date('2000/1/1').toISOString());
+    table.check(
```

Comment:
I think this check still allows both to be filled which also shouldnt be allowed.

Reply:
Added an extra constraint!